code/datums/components/material_container.dm

```diff
@@ -20,10 +20,14 @@
 	var/last_inserted_id
 	var/last_amount_inserted
 	var/last_insert_success
+	var/precise_insertion = FALSE
 	var/datum/callback/precondition
 	//MAX_STACK_SIZE = 50
 	//MINERAL_MATERIAL_AMOUNT = 2000
 
+/datum/component/material_container/precise
+	precise_insertion = TRUE
```

Comment:
Make it a parameter on Initialize instead of a subtype otherwise other material containers won't see it as a duplicate